src/Simulator/IncrementalRedstoneSimulator.h

```diff
@@ -348,47 +391,30 @@ class cIncrementalRedstoneSimulator :
 		}
 	}
 
-	inline static bool AreCoordsOnChunkBoundary(int a_BlockX, int a_BlockY, int a_BlockZ)
+	inline static bool DoesIgnorePlayerToggle(BLOCKTYPE a_Block)
 	{
-		return (  // Are we on a chunk boundary? +- 2 because of LinkedPowered blocks
-			((a_BlockX % cChunkDef::Width) <= 1) ||
-			((a_BlockX % cChunkDef::Width) >= 14) ||
-			((a_BlockZ % cChunkDef::Width) <= 1) ||
-			((a_BlockZ % cChunkDef::Width) >= 14)
-			);
-	}
-
-	inline static Vector3i AdjustRelativeCoords(const Vector3i & a_RelPosition)
-	{
-		if (
-			(a_RelPosition.x >= 0) && (a_RelPosition.x < cChunkDef::Width) &&
-			(a_RelPosition.z >= 0) && (a_RelPosition.z < cChunkDef::Width)
-			)
+		switch (a_Block)
 		{
-			return a_RelPosition;
+			case E_BLOCK_ACACIA_FENCE_GATE:
+			case E_BLOCK_BIRCH_FENCE_GATE:
+			case E_BLOCK_DARK_OAK_FENCE_GATE:
+			case E_BLOCK_FENCE_GATE:
+			case E_BLOCK_JUNGLE_FENCE_GATE:
+			case E_BLOCK_SPRUCE_FENCE_GATE:
+			case E_BLOCK_IRON_TRAPDOOR:
+			case E_BLOCK_TRAPDOOR:
+			{
+				return true;
+			}
+			default: return false;
 		}
+	}
 
-		Vector3i RelPos = a_RelPosition;
+	inline static std::vector<cChunk *> GetAdjacentChunks(const Vector3i & a_RelBlockPosition, cChunk * a_Chunk);
 
-		// Request for a different chunk, calculate chunk offset:
-		while (RelPos.x >= cChunkDef::Width)
-		{
-			RelPos.x -= cChunkDef::Width;
-		}
-		while (RelPos.x < 0)
-		{
-			RelPos.x += cChunkDef::Width;
-		}
-		while (RelPos.z >= cChunkDef::Width)
-		{
-			RelPos.z -= cChunkDef::Width;
-		}
-		while (RelPos.z < 0)
-		{
-			RelPos.z += cChunkDef::Width;
-		}
-
-		return RelPos;
+	inline static Vector3i AdjustRelativeCoords(const Vector3i & a_RelPosition)
+	{
+		return Vector3i((a_RelPosition.x % cChunkDef::Width + cChunkDef::Width) % cChunkDef::Width, a_RelPosition.y, (a_RelPosition.z % cChunkDef::Width + cChunkDef::Width) % cChunkDef::Width);
 	}
```

Comment:
Isn't it supposed to be ( x % y + y) % y? Since normal mod doesn't produce the wanted result on negative numbers (i.e. -5 % 16 should be 11 for relative adjustment but instead is -5), this is done.